src/main/java/homework_5/AppHW_5.java

```diff
@@ -0,0 +1,54 @@
+package homework_5;
+
+import homework_5.utils.MyReader;
+import homework_5.utils.StringChecker;
+import homework_5.utils.impl.MyReaderImpl;
+import homework_5.utils.impl.StringCheckerImpl;
+
+import java.util.NoSuchElementException;
+
+public class AppHW_5 {
```

Comment:
Opt: in general idea to separate user input class is a good one, but we should not forget that abstractions are made when they simplify reading and fixing the code, not make it less readable